app/models/relationship.rb

```diff
@@ -158,6 +158,7 @@ def create_or_update_parent_id
 
   def self.create_unless_exists(source_id, target_id, relationship_type, options = {})
     r = Relationship.where(source_id: source_id, target_id: target_id).where('relationship_type = ?', relationship_type.to_yaml).last
+    r = Relationship.where(target_id: target_id).last if r.nil?
```

Comment:
@melsawy, there is a unique index for the `target_id` column ([here](https://github.com/meedan/check-api/blob/develop/db/schema.rb#L622))... so, these two lines are redundant. I think we can replace these two lines (160 and 161) but just this:

```ruby
r = Relationship.where(target_id: target_id).last
```

Right?

And please fix the [failing test](https://github.com/meedan/check-api/actions/runs/13140070412/job/36664836781?pr=2201) (`RelationshipTest#test_should_create_relationship_unless_exists`).

Reply:
@caiosba there is a relationship_type condition; for example, if the same `source_id` and `target_id` exist but with different types, we should ignore this one and attempt to create a new relationship with the new type, as there is a [callback](https://github.com/meedan/check-api/blob/develop/app/models/relationship.rb#L23) before creating to destroy the existing one.
In this case, when the target_id is assigned to another source, I prefer to maintain two conditions but ensure that the target is not assigned to another source and swap the two conditions.
I'll push the new changes before re-request your review

Reply:
OK Sawy, makes sense!